contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,529 @@
+
+    // DefaultRaidGroupType is the default raid type which applies 
+    // to all the pools if raid type is not specified there
+    // Compulsory field if any raidGroup is not given Type
+    DefaultRaidGroupType string `json:"defaultRaidGroupType"`
+ 
+    // OverProvisioning to enable over provisioning
+    // Optional -- defaults to false
+    OverProvisioning bool `json:"overProvisioning"`
+
+    // Compression to enable compression
+    // Optional -- defaults to off
+    // Possible values : lz, off
+    Compression string `json:"compression"`
+}
+
+// RaidGroup contains the details of a raid group for the pool
+type RaidGroup struct{
+    // Type is the raid group type
+    // Supported values are : stripe, mirror, raidz and raidz2
+    
+    // stripe -- stripe is a raid group which divides data into blocks and 
+    // spreads the data blocks across multiple block devices.
+
+    // mirror -- mirror is a raid group which does redundancy
+    // across multiple block devices.
+
+    // raidz -- TODO
+   
+    // radiz2 -- TODO
+
+    // Optional -- defaults to `defaultRaidGroupType` present in `PoolConfig`
+    Type string `json:"type"`
+
+    // IsWriteCache is to enable this group as a write cache.
+    IsWriteCache bool `json:"isWriteCache"`
+ 
+    // IsSpare is to declare this group as spare which will be 
+    // part of the pool that can be used if some block devices 
+    // fail. 
+    IsSpare bool `json:"isSpare"`
+
+    // IsReadCache is to enable this group as read cache.
+    IsReadCache bool `json:"isReadCache"`
+
+    // BlockDevices contains a list of block devices that 
+    // constitute this raid group.
+    BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevices"`
+}
+
+// CStorPoolClusterBlockDevice contains the details of block devices that 
+// constitutes a raid group.
+type CStorPoolClusterBlockDevice struct{
+     // ID is system generated for the block device.
+     // Once assigned it never changes. One can consider
+     // this ID as a slot for this block device.
+     ID string `json:"ID"`
+
+     // BlockDeviceName is the name of the block device.
+     BlockDeviceName string `json:"blockDeviceName"`
+
+     // Capacity is the capacity of the block device.
+     // It is system generated
+     Capacity string `json:"capacity"`
+
+     // DevLink is the dev link for block devices
+     DevLink string `json:"devLink"`
+}
+     
+// CStorPoolClusterStatus is for handling status of pool.
+type CStorPoolClusterStatus struct {
+	Phase string `json:"phase"`
+}
+
+
+```
+
+#### Spec for CStorPool
+
+```
+apiVersion: openebs.io/v1alpha1
+kind: CStorPool
+metadata:
+  name: cstor-disk-pool-7hbs
+spec:
+  hostName: gke-cstor-it-default-pool-1
+  raidConfig:
+  - type: stripe
+    isWriteCache:false
+    isSpare: false
+    isReadCache: false
+    blockDevices:
+    - blockDeviceName: sparse-3c1fc7491f9e4cf50053730740647318
+      id: bd1
+      capacity: 12345234
+      devLink: "/dev/sda"
```

Comment:
`replaceBy` should be here at the blockDevice level

Reply:
`replaceBy` having at level parallel to `raidConfig` also fine as long as we have `id` in it